tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -1014,6 +1014,105 @@ ZTEST_USER(uart_async_long_buf, test_long_buffers)
+volatile static uint8_t __aligned(sizeof(void *))
+	short_rx_buf_pool[SHORT_RX_BUF_SIZE] __used __NOCACHE;
+#else
+volatile static ZTEST_BMEM uint8_t __aligned(sizeof(void *)) short_rx_buf[SHORT_RX_BUF_SIZE];
+volatile static ZTEST_BMEM uint8_t __aligned(sizeof(void *)) short_rx_buf_pool[SHORT_RX_BUF_SIZE];
+#endif /* NOCACHE_MEM */
+volatile static ZTEST_BMEM size_t short_rx_idx;
+volatile static ZTEST_BMEM size_t short_rx_pool_idx;
+volatile static ZTEST_BMEM size_t short_rx_buf_size;
+
+static void test_short_buffers_callback(const struct device *dev, struct uart_event *evt,
+					void *user_data)
+{
+	switch (evt->type) {
+	case UART_RX_RDY:
+		memcpy((void *)&short_rx_buf[short_rx_idx], &evt->data.rx.buf[evt->data.rx.offset],
+		       evt->data.rx.len);
+		short_rx_idx += evt->data.rx.len;
+		break;
+	case UART_RX_DISABLED:
+		k_sem_give(&rx_disabled);
+		break;
+	case UART_RX_BUF_REQUEST:
+		uart_rx_buf_rsp(dev, (uint8_t *)&short_rx_buf_pool[short_rx_pool_idx],
+				short_rx_buf_size);
+		short_rx_pool_idx += short_rx_buf_size;
+		break;
+	default:
+		break;
+	}
+}
+
+static void *short_buffers_setup(void)
+{
+	static int idx;
+
+	uart_async_test_init(idx++);
+
+	uart_callback_set(uart_dev, test_short_buffers_callback, NULL);
+
+	return NULL;
+}
+
+static void test_uart_async_short_buf(size_t buf_len, size_t tx_len)
+{
+	int ret;
+	uint8_t tx_buffer[SHORT_TX_BUF_SIZE];
+
+	for (size_t i = 0; i < SHORT_TX_BUF_SIZE; ++i) {
+		tx_buffer[i] = tx_len;
+	}
+
+	short_rx_idx = 0;
+	short_rx_pool_idx = 0;
+	memset((void *)short_rx_buf, 0, SHORT_RX_BUF_SIZE);
+	memset((void *)short_rx_buf_pool, 0, SHORT_RX_BUF_SIZE);
+
+	short_rx_buf_size = buf_len;
+
+	ret = uart_rx_enable(uart_dev, (uint8_t *)&short_rx_buf_pool[short_rx_pool_idx], buf_len,
+			     2 * USEC_PER_MSEC);
+	zassert_true(ret == 0, "[buff=%d][tx=%d]Failed to enable RX: %d\n", buf_len, tx_len, ret);
+	short_rx_pool_idx += buf_len;
+
+	ret = uart_tx(uart_dev, tx_buffer, tx_len, 100 * USEC_PER_MSEC);
+	zassert_true(ret == 0, "[buff=%d][tx=%d]Failed to TX: %d\n", buf_len, tx_len, ret);
+	k_msleep(10);
+
+	uart_rx_disable(uart_dev);
+	zassert_equal(k_sem_take(&rx_disabled, K_MSEC(500)), 0,
+		      "[buff=%d][tx=%d]RX_DISABLED timeout\n", buf_len, tx_len);
+
+	zassert_equal(short_rx_idx, tx_len,
+		      "[buff=%d][tx=%d]Wrong number of bytes received, got: %d, expected: %d\n",
+		      buf_len, tx_len, short_rx_idx, tx_len);
+	zassert_equal(memcmp((void *)short_rx_buf, tx_buffer, tx_len), 0,
+		      "[buff=%d][tx=%d]Buffers not equal\n", buf_len, tx_len);
+}
+
+ZTEST_USER(uart_async_short_buf, test_short_buffers)
+{
+	struct uart_config uart_cfg;
+
+	zassert_equal(uart_config_get(uart_dev, &uart_cfg), 0);
+	if (uart_cfg.baudrate > CONFIG_SHORT_BUFFER_TEST_BUADRATE_LIMIT) {
+		ztest_test_skip();
+	}
+
+	for (size_t buf_len = 1; buf_len < 10; ++buf_len) {
+		for (size_t tx_len = 1; tx_len < 12; ++tx_len) {
```

Comment:
```suggestion
		for (size_t tx_len = 1; tx_len < sizeof(SHORT_TX_BUF_SIZE); ++tx_len) {
```

and  SHORT_TX_BUF_SIZE can be limited.

Reply:
Done.